### PR TITLE
README.ES.md -> 01 & 02

### PR DESCRIPTION
**Pull request by zanezhub**
## New files 0353949
I translated the following files:
* README.ES.md ---> [01_wait_forever](https://github.com/rust-embedded/rust-raspberrypi-OS-tutorials/blob/master/01_wait_forever/README.md)
* README.ES.md ---> [02_runtime_init](https://github.com/rust-embedded/rust-raspberrypi-OS-tutorials/blob/master/02_runtime_init/README.md)

### CODE REVIEW

File: 01_wait_forever/README.ES.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Tutorial 01 - Esperar para siempre
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:1]:
```suggestion
# Tutorial 01 - Esperar infinitamente
```

File: 01_wait_forever/README.ES.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## tl;dr
+
+* Se configura la estructura que tendrá el proyecto.
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:5]:
```suggestion
* Se configura la estructura que tiene el proyecto.
```

File: 01_wait_forever/README.ES.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+* Se ejecuta una pequeño código hecho en ensamblador que tiene como función detener todos los núcleos del procesador que están ejecutando el kernel.
+
+## Compilar
+
+* El archivo `Makefile` selecciona:
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:11]:
```suggestion
* El archivo `Makefile` permite ejecutar:
```

File: 01_wait_forever/README.ES.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  
+  * `doc`: Genera la documentación.
+  
+  * `qemu`: Ejecutar el kernel en QEMU.
+  
+  * `clippy`
+  
+  * `clean`
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:19]:
```suggestion
  * `clippy`: Analiza el código y sugiere mejoras.
  
  * `clean`: Elimina todos los archivos generados durante la compilación, etc.
```

File: 01_wait_forever/README.ES.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  
+  * `readelf`: Inspeccionar la salida de `ELF`.
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:21]:
```suggestion
  * `readelf`: Inspecciona el archivo `ELF` de salida.
```

File: 01_wait_forever/README.ES.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  
+  * `objdump`: Inspecciona el ensamblador. 
+  
+  * `nm`: Inspecciona los símbolos.
+
+## Código a revisar
+
+* El script enlazador específico de `BSP` llamado `link.ld`.
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:29]:
```suggestion
* El script para enlazado específico para la `BSP` llamado `link.ld`.
```

File: 01_wait_forever/README.ES.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  
+  * Carga la dirección en `0x8_0000`.
+  
+  * Solo la sección `.text`.
+
+* `main.rs`: [Atributos internos](https://doc.rust-lang.org/reference/attributes.html) importantes:
+  
+  * `#![no_std]`, `#![no_main]`.
+
+* `boot.s`: La función de ensamblador `__start()` que inicia `wfe` (Wait For Event / Esperar Por Un Evento), detiene todos los núcleos del procesador que están ejecutando `_start()`.
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:39]:
```suggestion
* `boot.s`: La función de ensamblador `_start()` que inicia `wfe` (Wait For Event / Esperar Hasta Un Evento), detiene todos los núcleos del procesador que están ejecutando `_start()`. 
```

File: 01_wait_forever/README.ES.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  
+  * Solo la sección `.text`.
+
+* `main.rs`: [Atributos internos](https://doc.rust-lang.org/reference/attributes.html) importantes:
+  
+  * `#![no_std]`, `#![no_main]`.
+
+* `boot.s`: La función de ensamblador `__start()` que inicia `wfe` (Wait For Event / Esperar Por Un Evento), detiene todos los núcleos del procesador que están ejecutando `_start()`. 
+
+* Tenemos que definir una función `#[panic_handler]` para que el compilador no nos cause problemas.
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:41]:
```suggestion
* Tenemos que definir una función que funcione como `#[panic_handler]` (manejador de pánico) para que el compilador no nos cause problemas.
```

File: 01_wait_forever/README.ES.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  
+  * Hazla una `unimplemented!()` porque se eliminará ya que no está siendo usada.
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:43]:
```suggestion
  * Hazla `unimplemented!()` porque se eliminará ya que no está siendo usada.
```

File: 01_wait_forever/README.ES.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+## Pruébalo
+
+Dentro de la carpeta del proyecto, ejecuta a QEMU y mira el núcleo del procesador hilado en `wfe`:
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:47]:
```suggestion
Dentro de la carpeta del proyecto, ejecuta a QEMU y mira el núcleo del procesador ejecutando `wfe` en bucle:
```

File: 02_runtime_init/README.ES.md [hunk 11]
```diff
@@ -0,0 +1,348 @@
+# Tutorial 02 - Inicio de Tiempo de Ejecución
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:1]:
```suggestion
# Tutorial 02 - Inicialización del `runtime`
```

File: 02_runtime_init/README.ES.md [hunk 12]
```diff
@@ -0,0 +1,348 @@
...
+
+## tl;dr
+
+* Extendimos la funcionalidad de `boot.s` para que sea capaz de llamar código de Rust por primera vez. Antes de que el cambio a Rust ocurra, se realizan algunos trabajos de inicio de tiempo de ejecución (runtime init).
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:5]:
```suggestion
* Extendimos la funcionalidad de `boot.s` para que sea capaz de llamar código Rust por primera vez. Antes de que el cambio a Rust ocurra, se realizan algunos trabajos de inicialización del `runtime` (soporte para ejecución de código).
```

File: 02_runtime_init/README.ES.md [hunk 13]
```diff
@@ -0,0 +1,348 @@
...
+* El código de Rust que es llamado solo pausa la ejecución con una llamada a `panic!()`.
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:6]:
```suggestion
* El código Rust que es llamado solo pausa la ejecución con una llamada a `panic!()`.
```

File: 02_runtime_init/README.ES.md [hunk 14]
```diff
@@ -0,0 +1,348 @@
...
+* Ejecuta `make qemu` de nuevo para que puedas ver el código adicional en acción.
+
+## Adiciones importantes
+
+* Adiciones importantes al script `link.ld`:
+  
+  * Nuevas secciones: `.rodata`, `.got`, `.data`, `.bss`.
+  
+  * Un lugar totalmente dedicado para enlazar argumentos de tiempo de arranque (boot-time) que necesitan estar listos cuando se llame a `_start()`.
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:15]:
```suggestion
  * Un lugar totalmente dedicado a enlazar argumentos de tiempo de arranque (boot-time) que necesitan estar listos cuando se llame a `_start()`.
```

File: 02_runtime_init/README.ES.md [hunk 15]
```diff
@@ -0,0 +1,348 @@
...
+
+* `_start()` en `_arch/__arch_name__/cpu/boot.s`:
+  
+  1. Para el núcleo del procesador `if core != core0` (si la variable `core` es diferente de la variable `core0`).
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:19]:
```suggestion
  1. Para todos los núcleos excepto el núcleo 0.
```

File: 02_runtime_init/README.ES.md [hunk 16]
```diff
@@ -0,0 +1,348 @@
...
+  
+  2. Inicializa la [`DRAM`](https://es.wikipedia.org/wiki/DRAM) haciendo ceros la sección de [`bsp`](https://en.wikipedia.org/wiki/.bss).
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:21]:
```suggestion
  2. Inicializa la [`DRAM`](https://es.wikipedia.org/wiki/DRAM) poniendo a cero la sección de [`.bss`](https://en.wikipedia.org/wiki/.bss).
```

File: 02_runtime_init/README.ES.md [hunk 17]
```diff
@@ -0,0 +1,348 @@
...
+  
+  3. Configura el `stack pointer` (puntero a la memoria *stack*).
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:23]:
```suggestion
  3. Configura el `stack pointer` (puntero a la memoria pila).
```

File: 02_runtime_init/README.ES.md [hunk 18]
```diff
@@ -0,0 +1,348 @@
...
+  
+  4. Salta hacia la función `_start_rust()`, definida en `arch/__arch_name__/cpu/boot.rs`.
+
+. `_start_rust()`:
+  
+  . Llama a `kernel_init()`, que llama a `panic!()`, que eventualmente también pausa a core0.
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:29]:
```suggestion
  . Llama a `kernel_init()`, que llama a `panic!()`, que al final también pone al núcleo 0 en pausa.
```

File: 02_runtime_init/README.ES.md [hunk 19]
```diff
@@ -0,0 +1,348 @@
...
+
+. La librería ahora usa el crate [cortex-a](https://github.com/rust-embedded/cortex-a), que nos da abstracciones conocidas como *zero-overhead* y envulve las partes que hacen uso de un `unsafe` (partes con código que no es seguro y podría causar errores) cuando se trabaja con los recursos del procesador.
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:31]:
```suggestion
. La librería ahora usa el crate [cortex-a](https://github.com/rust-embedded/cortex-a), que nos da abstracciones sin coste y envuelve las partes que hacen uso de un `unsafe` (partes con código que no es seguro y podría causar errores) cuando se trabaja directamente con los recursos del procesador.
```

File: 02_runtime_init/README.ES.md [hunk 20]
```diff
@@ -0,0 +1,348 @@
...
+  
+  . Lo puedes ver en acción en `_arch/__arch_name__/cpu.rs`.
+
+## Diff del archivo anterior
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:35]:
```suggestion
## Diferencia con el archivo anterior
```

**Review by eldruin — COMMENTED** [review]
Thanks! Just added a few improvements/corrections.

**Comment by zanezhub** [comment]
I just made a new commit e073b53 with the corrections and improvements. Thankss

File: 01_wait_forever/README.ES.md [hunk 21]
```diff
@@ -1,32 +1,32 @@
-# Tutorial 01 - Esperar para siempre
+# Tutorial 01 - Esperar para infinitamente
```
Comment (eldruin) [inline 01_wait_forever/README.ES.md:1]:
```suggestion
# Tutorial 01 - Esperar infinitamente
```

File: 02_runtime_init/README.ES.md [hunk 22]
```diff
@@ -12,27 +12,27 @@
...
-  * Un lugar totalmente dedicado para enlazar argumentos de tiempo de arranque (boot-time) que necesitan estar listos cuando se llame a `_start()`.
+  * Un lugar totalmente dedicado a enlazar argumentos de tiempo de arranque (boot-time) que necesitan estar listos cuando se llame a `_start()`.
 
 * `_start()` en `_arch/__arch_name__/cpu/boot.s`:
   
-  1. Para el núcleo del procesador `if core != core0` (si la variable `core` es diferente de la variable `core0`).
+  1. Para todos los núcleos expecto el núcleo 0.
   
-  2. Inicializa la [`DRAM`](https://es.wikipedia.org/wiki/DRAM) haciendo ceros la sección de [`bsp`](https://en.wikipedia.org/wiki/.bss).
+  2. Inicializa la [`DRAM`](https://es.wikipedia.org/wiki/DRAM) poniendo a cero la sección de [`bss`](https://en.wikipedia.org/wiki/.bss).
```
Comment (eldruin) [inline 02_runtime_init/README.ES.md:21]:
```suggestion
  2. Inicializa la [`DRAM`](https://es.wikipedia.org/wiki/DRAM) poniendo a cero la sección [`.bss`](https://en.wikipedia.org/wiki/.bss).
```

**Review by eldruin — COMMENTED** [review]
Ok I fixed a couple that were left myself.

**Comment by eldruin** [comment]
Thank you!

**Pull request merged**